src/main/scala/rifs/business/tables/ApplicationTables.scala

```diff
@@ -119,6 +119,14 @@ class ApplicationTables @Inject()(dbConfigProvider: DatabaseConfigProvider)(impl
     }
   }
 
+  override def clearSectionCompletedDate(id: ApplicationId, sectionNumber: Int): Future[Int] = {
+    fetchAppWithSection(id, sectionNumber).flatMap {
+      case Some((app, Some(section))) => db.run(appSectionC(id, sectionNumber).update(section.copy(completedAt = null)))
+      //TODO: do we need a ormal approach for such failures ?
+      case _ => Future.failed(new Throwable(s"Section $sectionNumber in applicaton $id not found"))
```

Comment:
I'd say here you could return 0 to indicate no rows were updated and have the controller return a 404 response if that happens. Rr change the method signature to return `Future[Option[Int]]` and return `None` in this situation. Again the controller can then return a 404  response. Should definitely *not* be throwing an exception.